### PR TITLE
UCT/IB/RC: Scatter to CQE upstream support

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8361/ for details (Mellanox internal link).

File: src/uct/ib/rc/accel/rc_mlx5_iface.c [hunk 1]
```diff
@@ -179,6 +179,35 @@ static void uct_rc_mlx5_iface_progress_enable(uct_iface_h tl_iface, unsigned fla
...
+    if (qp == NULL) {
+        ucs_error("iface=%p: failed to create QP: %m", iface);
+        return UCS_ERR_IO_ERROR;
+    }
+
+    attr->cap = attr->ibv.cap;
+    *qp_p     = qp;
+
+    return UCS_OK;
+#else
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_iface.c]: do we set get.min_zcopy to 0, for uct_iface_query(), in case HAVE_DECL_MLX5DV_CREATE_QP is not defined?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8368/ for details (Mellanox internal link).

File: src/uct/ib/rc/accel/rc_mlx5_iface.c (hunk 1 above)
Comment (yosefe) [reply src/uct/ib/rc/accel/rc_mlx5_iface.c]: @Artemy-Mellanox let's also assert we do not get inline CQE if we had set get.min_zcopy to 0 (during poll_tx)

Comment (yosefe) [reply src/uct/ib/rc/accel/rc_mlx5_iface.c]: use uct_rc_txqp_completion_op to determine if the operation was get_zcopy

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8379/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8380/ for details (Mellanox internal link).

**Comment by Artemy-Mellanox** [comment]
@yosefe

**Pull request merged**